### PR TITLE
[DNM] *: format diff

### PR DESCRIPTION
**Pull request by BusyJay**
This pr tries to only format the changed lines rather than anything when make format.

Note that, there is a [bug](https://github.com/rust-lang-nursery/rustfmt/issues/1445) in rustfmt that unrelated comment will be also formatted.

### CODE REVIEW

File: travis-build/Makefile [hunk 1]
```diff
@@ -38,12 +38,10 @@ ${LOCAL_DIR}/bin/kcov:
 		make install; \
 	fi
 
-prepare-rustfmt:
-	curl -L https://github.com/tennix/rustfmt/releases/download/v0.6/rustfmt-${RUSTFMT_VERSION}-linux-amd64.tar.gz -o rustfmt-${RUSTFMT_VERSION}-linux-amd64.tar.gz && \
-	mkdir -p ${HOME}/.cargo/bin && tar xzf rustfmt-${RUSTFMT_VERSION}-linux-amd64.tar.gz -C ${HOME}/.cargo/bin --strip-components=1
-	# cargo install --vers =0.6.0 rustfmt || exit 0
+${HOME}/.cargo/bin/rustfmt:
+	cargo install -f rustfmt
```
Comment (siddontang) [inline travis-build/Makefile]: do we need to use a special version?

Comment (BusyJay) [inline travis-build/Makefile]: Only changes are formatted, so I think it's ok to just use the latest version.

**Comment by siddontang** [comment]
Do we need to close #1749 now? @overvenus

**Comment by tennix** [comment]
What about circle CI?

For shell script I suggest [shellcheck](https://github.com/koalaman/shellcheck) as linter.

**Comment by siddontang** [comment]
PTAL @overvenus @tennix

**Comment by tennix** [comment]
LGTM

**Comment by siddontang** [comment]
PTAL @overvenus

**Comment by BusyJay** [comment]
It seems that implementation of file lines is naive, let's close it for now and reopen when the feature becomes stable.

**Pull request closed without merging**